js/zonda.js

```diff
@@ -619,64 +527,570 @@ module.exports = class zonda extends Exchange {
+        //
+        //     {
+        //          id: 'df00b0da-e5e0-11e9-8c19-0242ac11000a',
+        //          t: '1570108958831',
+        //          a: '0.04776653',
+        //          r: '0.02145854',
+        //          ty: 'Sell'
+        //     }
+        //
+        const isFetchTrades = 'ty' in trade;
+        const timestamp = this.safeInteger2 (trade, 'time', 't');
+        const side = this.safeStringLower2 (trade, 'userAction', 'ty');
+        const wasTaker = this.safeValue (trade, 'wasTaker');
+        let takerOrMaker = undefined;
+        if (wasTaker !== undefined) {
+            takerOrMaker = wasTaker ? 'taker' : 'maker';
+        } else if (isFetchTrades) {
+            takerOrMaker = 'taker';
+        }
+        const priceString = this.safeString2 (trade, 'rate', 'r');
+        const amountString = this.safeString2 (trade, 'amount', 'a');
+        const feeCostString = this.safeString (trade, 'commissionValue');
+        const marketId = this.safeString (trade, 'market');
+        market = this.safeMarket (marketId, market, '-');
+        const symbol = market['symbol'];
+        let fee = undefined;
+        if (feeCostString !== undefined) {
+            const feeCurrency = (side === 'buy') ? market['base'] : market['quote'];
+            fee = {
+                'currency': feeCurrency,
+                'cost': feeCostString,
+            };
+        }
+        const order = this.safeString (trade, 'offerId');
+        // todo: check this logic
+        let type = undefined;
+        if (order !== undefined) {
+            type = order ? 'limit' : 'market';
+        }
+        return this.safeTrade ({
+            'id': this.safeString (trade, 'id'),
+            'order': order,
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'symbol': symbol,
+            'type': type,
+            'side': side,
+            'price': priceString,
+            'amount': amountString,
+            'cost': undefined,
+            'takerOrMaker': takerOrMaker,
+            'fee': fee,
+            'info': trade,
+        }, market);
+    }
+
+    async fetchOHLCV (symbol, timeframe = '1m', since = undefined, limit = undefined, params = {}) {
+        /**
+         * @method
+         * @name zonda#fetchOHLCV
+         * @description fetches historical candlestick data containing the open, high, low, and close price, and the volume of a market
+         * @param {string} symbol unified symbol of the market to fetch OHLCV data for
+         * @param {string} timeframe the length of time each candle represents
+         * @param {int|undefined} since timestamp in ms of the earliest candle to fetch
+         * @param {int|undefined} limit the maximum amount of candles to fetch
+         * @param {object} params extra parameters specific to the zonda api endpoint
+         * @returns {[[int]]} A list of candles ordered as timestamp, open, high, low, close, volume
```

Comment:
Actually we return floats here (except the timestamp), but yeah, I see this is a widespread "bug"

Reply:
what should we do in this case atm? should I amend smth in this PR?

Reply:
> Actually we return floats here (except the timestamp), but yeah, I see this is a widespread "bug"

we actually return numbers here, which can be changed with their type by specifying `exchange.number = String` for example

Reply:
@frosty00 ok, I've set it to `number` type.